From adae9c24146a77384aa350c9dd6c0102b2e46fcc Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 27 Dec 2017 15:22:27 +0200
Subject: [PATCH 2191/2241] net: mvpp2x: compact pp2x structure by reordering

Reordering u16 and pointer fields causes for u16+u16
grouping, eliminates gaps and as result compacts the struct size.
Compact size improves cache hit factor.

This compensates new u16 field introduced by aggregated queue
alignment.

Change-Id: Idee9b82363ffcce7d0d172fc0be1205d4c567384
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48176
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
index 1c61915..63d8bc0 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
@@ -551,8 +551,8 @@ struct mv_pp2x {
 	struct platform_device *pdev;
 
 	/* List of pointers to port structures */
-	u16 num_ports;
 	struct mv_pp2x_port **port_list;
+	u16 num_ports;
 
 	/* Aggregated TXQs */
 	u16 num_aggr_qs;
-- 
2.7.4

